simple_ldap_user/simple_ldap_user.module

```diff
@@ -626,3 +633,20 @@ function simple_ldap_user_variable_get($name, $default = NULL, $force_default =
 
   return variable_get($name, $default);
 }
+
+/**
+ * Return the user account control status under Active Directory in LDAP database
+ * Check whether the user is disabled (Active Directory only).
+ * http://support.microsoft.com/kb/305144
+ *
+ * parameter: $ldap_user
+ *  */
+function simple_ldap_useraccountcontrol_isblocked ($ldap_user) {
+  // Force to update drupal user status based on $ldap_user->useraccountcontrol
+  if ($ldap_user->server->type == 'Active Directory') {
+    if (isset($ldap_user->useraccountcontrol[0]) && (int) $ldap_user->useraccountcontrol[0] & 2) {
+      return 0; // blocked
+    }
+  }
+  return ($ldap_user->exists)? 1 : 0; // active:blocked
```

Comment:
By adding the check for `$ldap_user->exists` here, the logic of the method no longer matches the function documentation.  It also alters the conditions which were originally in place at lines 62 and following.

This function should provide a method for checking an existing user in the external directory for conditions which disallow login (such as ActiveDirectory's useraccountcontrol attribute).

The check for the `$ldap_user->exists` should be outside of this function.